docset/winserver2022-ps/hyper-v/Dismount-VMHostAssignableDevice.md

```diff
@@ -0,0 +1,218 @@
+---
+description: Dismounts an assigned device from a virtual machine host.
+external help file: Microsoft.HyperV.PowerShell.Cmdlets.dll-Help.xml
+Module Name: Hyper-V
+ms.date: 06/12/2024
+online version: https://learn.microsoft.com/powershell/module/hyper-v/dismount-vmhostassignabledevice?view=windowsserver2022-ps&wt.mc_id=ps-gethelp
+schema: 2.0.0
+title: Dismount-VMHostAssignableDevice
+---
+
+# Dismount-VMHostAssignableDevice
+
+## SYNOPSIS
+Dismounts a device from a virtual machine (VM) host.
+
+## SYNTAX
+
+```
+Dismount-VMHostAssignableDevice [-InstancePath <String>] [-LocationPath <String>] [-Force]
+ [-Passthru] [-CimSession <CimSession[]>] [-ComputerName <String[]>] [-Credential <PSCredential[]>]
+ [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+## DESCRIPTION
+
+The `Dismount-VMHostAssignableDevice` cmdlet is used to dismount an assignable device from the
+virtual machine host. Use this cmdlet when you need to reassign a device or resolve conflicts
+between devices and virtual machines.
+
+## EXAMPLES
+
+### Example 1
+
+```powershell
+Dismount-VMHostAssignableDevice -InstancePath "PCIROOT(0)#PCI(1D02)#PCI(0000)"
+```
+
+This example dismounts the device with the specified instance path from the host, making it
+available for reassignment to another virtual machine.
+
+## PARAMETERS
+
+### -CimSession
+
+Runs the cmdlet in a remote session or on a remote computer. Enter a computer name or a session
+object, such as the output of a [New-CimSession](/powershell/module/cimcmdlets/new-cimsession)
+or [Get-CimSession](/powershell/module/cimcmdlets/get-cimsession) cmdlet. The default is the
+current session on the local computer.
+
+```yaml
+Type: CimSession[]
+Parameter Sets: (All)
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -ComputerName
+
+Specifies one or more Hyper-V hosts on the virtual network adapters are to be retrieved. NetBIOS
+names, IP addresses, and fully qualified domain names are allowed. The default is the local
+computer. Use localhost or a dot (`.`) to specify the local computer explicitly.
+
+```yaml
+Type: String[]
+Parameter Sets: Path
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Credential
+
+Specifies one or more user accounts that have permission to perform this action. The default is the
+current user.
+
+```yaml
+Type: PSCredential[]
+Parameter Sets: (All)
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Force
+
+Forces the command to run without asking for user confirmation.
```

Comment:
`-Force` also overrides some security checks in the platform, see also: https://learn.microsoft.com/en-us/windows-server/virtualization/hyper-v/deploy/deploying-graphics-devices-using-dda#install-the-partitioning-driver-optional

> Install the partitioning driver (optional)
DDA gives hardware vendors the ability to provide a security mitigation driver with their devices. This driver isn't the same as the device driver installed in the guest VM. It's up to the hardware vendor's discretion to provide this driver. But if they do provide a driver, install it before dismounting the device from the host partition. Reach out to the hardware vendor to see if they have a mitigation driver.

If no partitioning driver is provided, during dismount you must use the -Force option to bypass the security warning. For more information about the security implications, see [Plan for Deploying Devices using Discrete Device Assignment](https://learn.microsoft.com/en-us/windows-server/virtualization/hyper-v/plan/plan-for-deploying-devices-using-discrete-device-assignment).